app/models/profile.rb

```diff
@@ -17,6 +17,12 @@ class Profile < ApplicationRecord
 
   delegate :email, to: :user
 
+  scope :search_profiles, (lambda do |search_query|
+    left_outer_joins(:subjects)
+      .where('first_name ILIKE :q OR last_name ILIKE :q OR subjects.name ILIKE :q', q: "%#{search_query}%")
+      .group('profiles.id')
```

Comment:
add `unless search_query.blank?` in order to not apply scope if `search_query` didn't specified